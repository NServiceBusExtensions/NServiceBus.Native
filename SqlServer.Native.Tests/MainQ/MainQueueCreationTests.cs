using ApprovalTests;
using NServiceBus.Transport.SqlServerNative;
using Xunit;

public class MainQueueCreationTests
{
    [Fact]
    public void Run()
    {
        DbSetup.Setup();
        using (var connection = Connection.OpenConnection())
        {
            var manager = new QueueManager("MainQueueCreationTests", connection);
            manager.Drop().Await();
            manager.Create().Await();
            var sqlScriptBuilder = new SqlScriptBuilder(tables: true, namesToInclude: "MainQueueCreationTests");
            Approvals.Verify(sqlScriptBuilder.BuildScript(connection));
        }
    }
}