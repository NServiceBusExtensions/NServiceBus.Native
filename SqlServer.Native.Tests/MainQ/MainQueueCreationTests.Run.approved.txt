CREATE TABLE [dbo].[MainQueueCreationTests](
	[Id] [uniqueidentifier] NOT NULL,
	[CorrelationId] [varchar](255) COLLATE Latin1_General_CI_AS NULL,
	[ReplyToAddress] [varchar](255) COLLATE Latin1_General_CI_AS NULL,
	[Recoverable] [bit] NOT NULL,
	[Expires] [datetime] NULL,
	[Headers] [nvarchar](max) COLLATE Latin1_General_CI_AS NOT NULL,
	[BodyString]  AS (CONVERT([varchar](max),[Body])),
	[Body] [varbinary](max) NULL,
	[RowVersion] [bigint] IDENTITY(1,1) NOT NULL
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
